src/f3dasm/_src/datageneration/datagenerator.py

```diff
@@ -43,15 +44,32 @@ class DataGenerator:
     """Base class for a data generator"""
 
     def pre_process(self, experiment_sample: ExperimentSample, **kwargs) -> None:
-        """Function that handles the pre-processing"""
+        """Interface function that handles the pre-processing of the data generator
+
+        Notes
+        -----
+        If not implemented the function will be skipped
+        """
         ...
 
+    @abstractmethod
     def execute(self, **kwargs) -> None:
-        """Function that calls the FEM simulator the pre-processing"""
-        raise NotImplementedError("No execute function implemented!")
+        """Interface function that handles the execution of the data generator
+
+        Raises
+        ------
+        NotImplementedError
+            If the function is not implemented by the user
+        """
```

Comment:
Please make clear in the method docstring that the ExperimentSample is accessed as an attribute of the DataGenerator class and should be updated in-place.

Reply:
I have added this information as a note to the docstring